File: test/scripts/jl/test-script-load-font.jl
# examples/random-walkers-1.jl
using Pkg
Pkg.activate(joinpath(@__DIR__, ".."))
import SDLProcessing
const P = SDLProcessing
import SimpleDirectMediaLayer
const SDL2 = SimpleDirectMediaLayer.LibSDL2

## --- Sketch Definition ---
P.onsetup() do
    P.create_window(800, 600; title="Random Walkers")
    # do nothing
    P.textFont(P.asset_path("Open_Sans", "OpenSans-VariableFont_wdth,wght.ttf"), 24)
end

P.ondraw() do
    # do nothing
    P.background(P.color(0, 0, 0)) # Black background
    P.text("frameRate: $(round(Int, P.frameRate()))", 10, 10)
end

## --- Run the Sketch ---
P.run_sketch()

File: test/scripts/js/test-script-load-font.js
// Description: Test script for loading a font and displaying text.
// Based on: test/scripts/jl/test-script-load-font.jl

let myFont;

function preload() {
  // Path is relative to the HTML file.
  myFont = loadFont('../../../assets/Open_Sans/OpenSans-VariableFont_wdth,wght.ttf');
}

function setup() {
  createCanvas(800, 600);
  textFont(myFont);
  textSize(24);
}

function draw() {
  background(0); // Black background
  fill(255); // White text
  text(`frameRate: ${round(frameRate())}`, 10, 30);
}
